XamarinApp.Test/ViewModels/ApiDataPageViewModelTest.cs

```diff
@@ -0,0 +1,68 @@
+using AutoFixture;
+using Moq;
+using Prism.Navigation;
+using Prism.Services;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using XamarinApp.Models;
+using XamarinApp.Services.Interfaces;
+using XamarinApp.ViewModels;
+using Xunit;
+using static XamarinApp.Models.ApiModel;
+
+namespace XamarinApp.Test.ViewModels
+{
+  public class ApiDataPageViewModelTest
+  {
+    private readonly Mock<INavigationService> _navigationService;
+    private readonly Mock<IPageDialogService> _pageDialogService;
+    private readonly Mock<IRandomApiService> _randomApiService;
+    private readonly ApiDataPageViewModel viewModel;
+    private readonly Fixture _fixture = new Fixture();
+
+    public ApiDataPageViewModelTest()
+    {
+      _navigationService = new Mock<INavigationService>();
+      _pageDialogService = new Mock<IPageDialogService>();
+      _randomApiService = new Mock<IRandomApiService>();
+      viewModel = new ApiDataPageViewModel(_navigationService.Object, _pageDialogService.Object, _randomApiService.Object);
+    }
+
+    [Fact]
+    public void Will_Save_RandomApiService_Data_to_ApiData()
+    {
+      //Arrange
+      var apiData = _fixture.Build<ApiModel.Result>().CreateMany(50).ToList();
+      var data = new NavigationParameters();
+
+      //Act
+      _randomApiService.Setup(x => x.getRandomApiData()).ReturnsAsync(apiData);
+      viewModel.OnNavigatedTo(data);
+
+      //Assert
+      Assert.Equal(apiData, viewModel.ApiData);
+    }
+
+    [Fact]
+    public void Will_Navigate_to_DetailedPage()
+    {
+      //Arrange
+      var apiData = _fixture.Build<ApiModel.Result>().CreateMany(50).ToList();
+      ObservableCollection<Result> data = new ObservableCollection<Result>(apiData as List<Result>);
+      var list = apiData.Where(x => x ==  apiData.FirstOrDefault()).ToList();
+      var specificData = list.FirstOrDefault();
+
+      //Act
+      viewModel.ApiData = data;
```

Comment:
this is not how you do this. this should be done by calling the OnNavigatedTo of the page so that It will fill the data in the collection

Reply:
Resolved